sdks/java/harness/src/test/java/org/apache/beam/fn/harness/state/StateBackedIterableTest.java

```diff
@@ -269,7 +269,8 @@ public void testByteObservingStateBackedIterable() throws Exception {
               .sum();
       observer.advance();
       // 5 comes from size and hasNext (see IterableLikeCoder)
-      assertEquals(iterateBytes + 5, observer.total);
+      // observer receives scaled
+      assertTrue(iterateBytes + 5 >= observer.total);
```

Comment:
is it less than 10 elements? shoudln't it be exact in that case?

If it's off by some double-math errors you could use some fuzzy equals (https://guava.dev/releases/23.0/api/docs/com/google/common/math/DoubleMath.html#fuzzyEquals-double-double-double- for example but perhaps some testing assert variant exists)

Reply:
it's more than 10 elements, added assert with tolerance.